mill_local/__init__.py

```diff
@@ -22,81 +39,95 @@ def __init__(self, device_ip, websession, timeout=15):
-                return response.status
+        _LOGGER.debug("Setting target temperature to: '%s'", target_temperature)
+        return await self._post_request(
+            command="set-temperature",
+            payload={
+                "type": "Normal",
+                "value": target_temperature,
+            }
+        )
 
-    async def set_normal_operation_mode(self):
-        """Set target temperature."""
-        payload = {"mode": "Control individually"}
-        with async_timeout.timeout(self._timeout):
-            async with self.websession.post(
-                f"{self.url}/operation-mode",
-                payload=payload,
-            ) as response:
-                _LOGGER.debug("Heater response %s", response.status)
-                if response.status != 200:
-                    _LOGGER.error(
-                        "Failed to set target temperature %s %s",
-                        response.status,
-                        response.reason,
-                    )
-                return response.status
+    async def set_operation_mode_control_individually(self):
+        """Set operation mode to 'control individually'."""
+        return await self._set_operation_mode(OperationMode.CONTROL_INDIVIDUALLY)
+
+    async def set_operation_mode_off(self):
+        """Set operation mode to 'off'."""
+        return await self._set_operation_mode(OperationMode.OFF)
 
     async def connect(self):
         """Get heater status."""
         return await self.get_status()
 
     async def get_status(self):
-        """Get heater status."""
+        """Get summary of the heater device information."""
         try:
-            self._status = await self._request("status")
+            self._status = await self._get_request("status")
         except (aiohttp.client_exceptions.ClientError, asyncio.TimeoutError):
             _LOGGER.error("Failed to get status", exc_info=True)
             return None
         return self._status
 
     async def fetch_heater_and_sensor_data(self):
-        """Get heater status."""
-        return await self._request("control-status")
+        """Get current heater state and control status."""
+        return await self._get_request("control-status")
+
+    async def _set_operation_mode(self, mode: OperationMode):
+        """Set heater operation mode."""
+        _LOGGER.debug("Setting operation mode to: '%s'", mode.value)
+        return await self._post_request(command="operation-mode", payload={"mode": mode.value})
+
+    async def _post_request(self, command: str, payload: dict):
+        """HTTP POST request to Mill Local Api."""
+        with async_timeout.timeout(self._timeout):
+            async with self.websession.post(
+                    url=f"{self.url}/{command}",
+                    data=json.dumps(payload),
+            ) as response:
+                _LOGGER.debug("POST '%s' response status: %s", command, response.status)
+                res = await response.json()
```

Comment:
Not 100% sure, but you are right that. We should maybe check the value of `response.status` before decoding the `response` to `json`. 

Note that json decoding might fail for other reasons as well, even `response.status `is `200`, and it will throw exception (see [doc](https://docs.aiohttp.org/en/stable/client_quickstart.html#json-response-content)). 

So it might be safer to wrap in a try/except block?

Same guarding should be done in the `_get_request()` function.

The original `_request()` did neither check for response.status, but `get_status()` wrapped the function call in a try/except. I think its cleaner to do this checking and try/except wrappings within the `*_request()` functions.